azurelinuxagent/ga/exthandlers.py

```diff
@@ -510,27 +519,30 @@ def handle_ext_handlers(self, goal_state_id):
         depends_on_err_msg = None
         extensions_enabled = conf.get_extensions_enabled()
 
-        # Instantiate policy engine, and use same engine to handle all extension handlers. If an error is thrown during
-        # policy engine initialization, we block all extensions and report the error via handler status for each extension.
-        # Save policy to history folder.
+        # Instantiate policy engine. If an error is thrown during policy engine initialization, we block all extensions
+        # and report the error via handler status for each extension.
         policy_error = None
         try:
-            gs_history = self.protocol.get_goal_state().history
-            policy_engine = ExtensionPolicyEngine()
-            if policy_engine is not None and policy_engine.policy_file_contents is not None and gs_history is not None:
-                gs_history.save(policy_engine.policy_file_contents, "waagent_policy.json")
-
+            self._policy_engine = ExtensionPolicyEngine()
         except Exception as ex:
             policy_error = ex
 
+        # Save policy contents to history folder. If there is an error saving to history, log a warning but continue processing extensions.
+        try:
+            gs_history = self.protocol.get_goal_state().history
+            if self._policy_engine is not None and self._policy_engine.policy_file_contents is not None and gs_history is not None:
```

Comment:
i believe we wanted the policy in the history folder to debug syntax errors, etc? if ExtensionPolicyEngine() raises. self._policy_engine will be None